### PR TITLE
Support arbitrary ammosets for siege artillery

### LINKED ISSUE #3269: [Bug]: Raider artillery buildable by players
**Specifications**

Rimworld version: 1.5 
Combat Extended version: 15.6.0.0
Combat Extended source (Steam, GitHub, etc.):GitHub
Your operating system: Win 11
Your mod list: https://rentry.co/wa32nkex

**Description**
Raider artillery buildable by players

**Expected behavior**
Raider artillery buildable only by NPCs and hidden to players

**To reproduce**
Research Mortars

**Screenshots & log dumps**

**Complete the following checklist**
I hereby verify that I have done the following:
- [x] Confirmed that my game version and load order are correct.
- [x] Confirmed that I am running the appropriate and most updated version of Combat Extended and required compatibility patches.
- [x] Confirmed I am not running any mods with known incompatibilities with Combat Extended.
- [x] Disabled Combat Extended and attempted to reproduce the behavior without success.

### PR DESCRIPTION
**Pull request by mszabo-wikia**
## Additions

- Support for arbitrary ammosets for siege artillery

## Changes

The vanilla logic around sieges is fairly inflexible -- it mostly assumes that only 81mm mortars are used, which effectively prevents using modded artillery in sieges without hacks like the "raider artillery" in our VFE:S patch that consumes the 81mm ammoset.

So, rework our patches around sieges to support artillery pieces with arbitrary ammosets, and remove raider artillery.

## References

- Closes #3269

## Reasoning
The raider artillery was only ever a stopgap, and it'll be nice to have the flexibility to use any artillery type and ammunition in sieges.

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] Playtested a colony -- tested spawning sieges with VFE:S, with a mixture of mortars and 155mm artillery

### CODE REVIEW

**Review by N7Huntsman — APPROVED** [review]
Works correctly in my testing using VFE: Security, and it's dropping the appropriate ammo types.

File: Source/CombatExtended/CombatExtended/SiegeUtility.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    public class SiegeUtility
+    {
+        /// <summary>
+        /// The minimum required construction skill to be able to build any potential siege artillery in the game.
+        /// </summary>
+        public static readonly int MinRequiredConstructionSkill;
+
+        static SiegeUtility()
+        {
+            MinRequiredConstructionSkill = DefDatabase<ThingDef>.AllDefsListForReading
```
Comment (N7Huntsman) [inline Source/CombatExtended/CombatExtended/SiegeUtility.cs:19]: Out of curiosity, how does vanilla handle making sure they spawn at least one pawn with high enough Construction to build the mortar? Do they have a dynamic check, or do they just make sure at least one has 5 construction (which is what mortars require)?

**Review by N7Huntsman — APPROVED** [review]

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
With all DLCs loaded, everything works as intended (see my other review).

However, somewhere in here we've baked-in a dependency upon Biotech. There are no errors thrown if you play without Biotech, but sieges won't be supplied with shells.

**Review by N7Huntsman — APPROVED** [review]
After some adjustments to the checks, I'm satisfied with the state of things, and everything works as intended both with and without DLCs active.

**Review by Thorbane — APPROVED** [review]

**Pull request merged**